### ISSUE TITLE
AssistedInject like in Guice DI

### ISSUE THREAD

**Issue by funkrusher** [quarkusio/quarkus#39730]
### Description

Guice DI provides a Feature called "Assisted Inject". See: https://github.com/google/guice/wiki/AssistedInject
With this feature it is possible to Create Beans with a combination of Injection-Parameters and Constructor-Parameters.

In a SaaS-Application, we usually have REST-Endpoints in the Quarkus-App, which are executed and called with data:
- Data from Authentication/Authorization (SecurityIdentity)
- Parameters given by the caller.

For example:

```
    @Inject
    ProductManager productManager;

    @Inject
    FkSecurityIdentity fkSecurityIdentity;

    @GET
    @Authenticated
    @Operation(summary = "returns the product with the specified id")
    @APIResponse(responseCode = "200", description = "Getting the product with the specified id successful")
    @APIResponse(responseCode = "500", description = "Server unavailable")
    @Path("/{productId}")
    public ProductDTO getOne(Long productId) throws NotFoundException {
        RequestContext request = new RequestContext(fkSecurityIdentity, 1);
        return productService.getOne(request, productId).orElseThrow(NotFoundException::new);
    }
```

If i would be able to use "Assisted Inject" in Quarkus, i would be able to create the `ProductManager` instance by passing it parameters from the REST-Request (SecurityIdentity, ...) as constructor arguments (@Assisted) to the ProductManager, while at the same time, being able to pass other objects to the ProductManager via @Inject.

This would allow me to combine the benefits of both worlds.

**Comment by mkouba** [comment]
> I could not find a similar concept in Quarkus ArC.

Yes, in ArC (and CDI in general) you would need a handwritten factory bean to create a _non-contextual_ object with custom arguments (`RequestContext request` and `Long productId` from your example). And of course, this factory bean would have to inject the CDI dependencies.

I personally don't see much benefit in this feature (at least in the current form of AssistedInject) but we could try to draw some use cases and if we find it useful we could continue with prototyping API etc.

**Comment by hu-chia** [comment]
There is an [AutoFactory](https://github.com/google/auto/tree/main/factory), in my opinion, It's a better and readable `AssistedInject`.
